### ISSUE TITLE
Rendering bug with symbolicnames

### ISSUE THREAD

**Issue by K-Lovelace** [neo4j-contrib/cypher-dsl#197]
Hi there,

We encountered some critical bugs on 2021.2.2 while rendering queries.

Here is what I could reproduce : 

```java
Node node = node("Division").named("node"), topNode = node("Division").named("topNode");
        Statement q = match(node)
            .withDistinct(node)
            .where(not(anyNode(node.getRequiredSymbolicName()).relationshipTo(node("Department"), "IN")
                                                              .relationshipTo(node("Department"), "INSIDE")
                                                              .properties("rel_property", literalTrue())
                                                              .relationshipTo(node("Employee"), "EMPLOYS")
            )).returning(asterisk()).build();
        // This works as expected (this is what we test and check to prevent any breaking changes)
        assertEquals("MATCH (node:Division)\n" +
                         "WITH DISTINCT node\n" +
                         "WHERE NOT (node)-[:IN]->(:Department)-[:INSIDE  {\n" +
                         "  rel_property: true\n" +
                         "}]->(:Department)-[:EMPLOYS]->(:Employee)\n" +
                         "RETURN *", Renderer.getRenderer(Configuration.prettyPrinting()).render(q));
       // This fails, and it's a problem because that's what we send to the database to be executed
        assertEquals(
            "MATCH (node:`Division`) WITH DISTINCT node WHERE NOT (node)-[:`IN`]->(:`Department`)-[:`INSIDE` {rel_property: true}]->(:`Department`)-[:`EMPLOYS`]->(:`Employee`) RETURN *",
            Renderer.getDefaultRenderer().render(q));
```

As you can see, the second rendering still introduces automatically generated names in the Condition. But there's also a problem where the pretty-printed cypher and the normally-printed cypher behave differently, which is our main concern here.

As for the critical bug, I could'nt reproduce it without our whole cypher-generating project, but the same where (`NOT (node)-[....` ended up as ` NOT ()-[....`, leading to a never-ending query at runtime. It's stable enough on 2021.2.1, although on this tag the naming problem is reversed (pretty-printing adds a lot of names that are not in the normal-printed cypher).

Please tell me if I am clear enough.

**Comment by michael-simons** [comment]
The pretty printing renderer triggers the name resolution. if you omit the pretty printed test, it works. But I see to fix this nevertheless.

**Comment by michael-simons** [comment]
This is definite a bug.

It can be mitigated by reusing your `node`:

```java
Statement q = Cypher.match(node)
			.withDistinct(node)
			.where(Conditions
				.not(node.relationshipTo(Cypher.node("Department"), "IN")
					.relationshipTo(Cypher.node("Department"), "INSIDE")
					.properties("rel_property", Cypher.literalTrue())
					.relationshipTo(Cypher.node("Employee"), "EMPLOYS")
				)).returning(Cypher.asterisk()).build();
```

**Comment by michael-simons** [comment]
Please see my commit message in the above linked commit. This is fixed.
Thanks for your great contributions, this is much appreciated.

**Comment by michael-simons** [comment]
Let me know if this 

> As for the critical bug, I could'nt reproduce it without our whole cypher-generating project, but the same where (NOT (node)-[.... ended up as NOT ()-[...., leading to a never-ending query at runtime. It's stable enough on 2021.2.1, although on this tag the naming problem is reversed (pretty-printing adds a lot of names that are not in the normal-printed cypher).

Is fixed. Hard to parse and reproduce.

**Comment by michael-simons** [comment]
With 2021.2.2 you might be as well affected by #199, which will be fixed in 2021.2.3. We will release this shortly.

**Comment by K-Lovelace** [comment]
Thanks for the quick answer, I'll try and test that as soon as I can and try to reproduce the bug I mentioned. For now we rolled back to 2021.1.1, we have worked around the different problems.

**Comment by michael-simons** [comment]
Hey :) 

Fixed and released: https://github.com/neo4j-contrib/cypher-dsl/releases/tag/2021.2.3
